crates/bevy_ui/src/focus.rs

```diff
@@ -166,10 +166,10 @@ pub fn ui_focus_system(
             FocusPolicy::Pass => { /* allow the next node to be hovered/clicked */ }
         }
     }
-    // reset lower nodes to None
+    // reset lower nodes to None, except if they're clicked
     for (_entity, _focus_policy, interaction, _) in moused_over_z_sorted_nodes {
         if let Some(mut interaction) = interaction {
-            if *interaction != Interaction::None {
+            if *interaction != Interaction::None && *interaction != Interaction::Clicked {
```

Comment:
I was thinking of `*interaction == Interaction::Hovered` instead here since those are the only 3 states, but it may be less obvious what it means then. Let me know if you think I should change it.

Reply:
It actually feels like this is two seperate if statements to me, each with their own purpose.

The !Clicked condition is to fix the bug, but the !None condition is to avoid spuriously triggereing change detection.

Reply:
Is it better now?